client_src/js/lib/octofarm_client.js

```diff
@@ -0,0 +1,42 @@
+export default class OctoFarmClient {
+  static async get(item) {
+    const url = `/${item}`;
+    const get = await fetch(url, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json"
+      }
+    });
+    if (get.status === 400) {
+      throw new Error(
+        `Malformed request! Status: ${get.status} - ${get.statusText}`
+      );
+    } else if (get.status === 503) {
+      throw new Error("Error contacting server, is it alive?");
+    } else if (get.status === 204) {
+      return;
+    }
+    return await get.json();
+  }
+  static async post(item, data) {
+    const url = `/${item}`;
+    const post = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    if (post.status === 400) {
+      throw new Error(
+        `Malformed request! Status: ${post.status} - ${post.statusText}`
+      );
+    } else if (post.status === 503) {
+      throw new Error("Error contacting server, is it alive?");
```

Comment:
Are we ready for these errors to be thrown? Javascript breaks and quits on the first errors it sees.

Reply:
I try catch everywhere I call that. And plan to do so for the rest I migrate over.  I'd love a global error handler but it's not something I'd say was worth our time pending V2. Or, could be done in another PR as I do want to migrate the client over to this file but the codes a reet spaghetti junction aha.

Reply:
I gather I can move the try and catch inside this file and handle it there? We would have a global OctoFarm connection error handler then?

Reply:
Wouldn't need to repeat it around the rest of the files either. Let me know will update it.

Reply:
I always think of throw new Error, if caught to continue with the rest of the code. If this response is from the server then it's considered an error to me. Should this not be the case?

Reply:
All Im saying is that I spotted new code with an explicit `throw new Error(...)` which if thrown could break the UI.
If you are certain that we try-catch everything, sure no prob. I dont oversee that.

Oh note that code 503 and 400 are rare right now (400 is rarely thrown because we dont have input validation everywhere, its why I proposed using `npm-input-validation` as shown in octoprint-companion PR). 
Codes 0, 500, 404 are more common and I dont see all of those covered.

Oh and why an else on 204 if it doesnt do anything but return?

Reply:
There's an else on 204 because I use 204 as a confirmation response from the server where no return data is needed. The return the function to let the rest skip which is at the bottom converting anything else to a JSON object. There would be nothing to convert if it didn't return and stop the .JSON.